library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -17,29 +17,60 @@
  * Sections are displayed in the same order they were added.
  */
 public class SectionedRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
-
+    
     public final static int VIEW_TYPE_HEADER = 0;
     public final static int VIEW_TYPE_FOOTER = 1;
     public final static int VIEW_TYPE_ITEM_LOADED = 2;
     public final static int VIEW_TYPE_LOADING = 3;
     public final static int VIEW_TYPE_FAILED = 4;
     public final static int VIEW_TYPE_EMPTY = 5;
 
+    private final static int mCustomViewTypeStart = 200;
+    private HashMap<Integer, CustomViewType> mCustomViewTypes;
+
     private LinkedHashMap<String, Section> sections;
     private HashMap<String, Integer> sectionViewTypeNumbers;
     private int viewTypeCount = 0;
     private final static int VIEW_TYPE_QTY = 6;
 
+    public SectionedRecyclerViewAdapter(CustomViewType...customViewTypes) {
+        sections = new LinkedHashMap<>();
+        sectionViewTypeNumbers = new HashMap<>();
+        mCustomViewTypes = new HashMap<>();
+        for (CustomViewType customViewType : customViewTypes) {
+            int key = mCustomViewTypeStart + mCustomViewTypes.size();
+            mCustomViewTypes.put(key, customViewType);
+        }
+    }
+
     public SectionedRecyclerViewAdapter() {
         sections = new LinkedHashMap<>();
         sectionViewTypeNumbers = new HashMap<>();
+        mCustomViewTypes = new HashMap<>();
+    }
+
+    public int getCustomViewTypeKey(CustomViewType customViewType) {
+        for (Map.Entry<Integer, CustomViewType> entry : mCustomViewTypes.entrySet()) {
+            if (entry.getValue() == customViewType) {
+                return entry.getKey();
+            }
+        }
+        return -1;
```

Comment:
[CODE] Should throw an exception - unless it's impossible to the user to know that this customViewType doesn't have a key